components/ruby/lib/chef-licensing.rb

```diff
@@ -66,5 +66,9 @@ def client(opts = {})
     def add_license
       ChefLicensing::LicenseKeyFetcher.add_license
     end
+
+    def warn_if_non_commercial_license(opts = {})
```

Comment:
Instead of creating a new API, can we not make this part of fetch and persist itself? That way we will not require any change on InSpec.

Reply:
But I see are we doing this because we explicitly need to check for cli reporter?

Reply:
But again we show all kinds of chef-licensing prompts in all reporters today. Having this check for cli reporter explicitly does not make sense. Thoughts?

Reply:
As discussed, we are going to display warning for any given commands, and not just specific to `exec` with `cli` reporter.